### PR TITLE
[AMSDK-9006] Updates to Identity docs to include more event details.

### PR DESCRIPTION
**Pull request by samthiru**

### CODE REVIEW

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 1]
```diff
@@ -4,8 +4,160 @@
 
 The following events are handled by the Identity extension:
 
+### Analytics Identity Response
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: We should move these events after the Identity request Identity and Generic Identity request events.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 2]
```diff
@@ -4,8 +4,160 @@
...
+
+#### Event Details
+
+| Type                          | Source                                 | Paired | Direction                                                    |
+| :---------------------------- | :------------------------------------- | :----- | :----------------------------------------------------------- |
+| com.adobe.eventType.analytics | com.adobe.eventSource.responseIdentity | True   | This is the end of the paired [Event: v5 Analytics Request Identity](https://wiki.corp.adobe.com/display/ADMSMobile/Event%3A+v5+Analytics+Request+Identity) event containing the AID value |
+
+#### Event Description
+
+This event is a response from the Analytics module that contains the unique tracking identifier. This value may be null if identity module is enabled and there is a valid MID generated.
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: This event is a response from the Analytics extension ...
... may be null if the Identity extension is enabled and there is a valid ECID generated.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 3]
```diff
@@ -4,8 +4,160 @@
...
+
+#### Generators
+
+This event will be generated by:
+
+- calling the Analytics module upon processing the com.adobe.eventSource.requestIdentity event.
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Please verify the Analytics extension to verify when this event is generated.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 4]
```diff
@@ -4,8 +4,160 @@
...
+| Key  | Value Type | Optional | Description                             |
+| :--- | :--------- | :------- | :-------------------------------------- |
+| aid  | String     | No       | The analytics tracking identifier value |
+
+#### Event Data Example
+
+**sample**
+
+```
+`{``    ``"aid" : "aidValue"``}`
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Formatting: 
```
{"aid" : "aidValue"}
```

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 5]
```diff
@@ -4,8 +4,160 @@
 
 The following events are handled by the Identity extension:
 
+### Analytics Identity Response
+
+#### Event Details
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Let's use the following names & order for all events:
Event name 
Include a short description & when it is generated
**Event details**
**Data payload definition**
**Event data example**

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 6]
```diff
@@ -4,8 +4,160 @@
...
+This event will be generated in the following scenarios:
+
+- When a request is made to update the Audience manager profile 
+- When a request is made to retrieve the current Audience manager profile 
+
+#### Event Details
+
+| Type                                | Source                                | Paired | Direction                                                    |
+| :---------------------------------- | :------------------------------------ | :----- | :----------------------------------------------------------- |
+| com.adobe.eventType.audienceManager | com.adobe.eventSource.responseContent | True   | This is the response to a paired event. This should be published both with and without a paired ID |
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: We can remove the extra detail about paired id as for third party developers there is no concept of pairId. 
You can include a link to the Audience response content event https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-audience-manager/audience-manager-event-reference#audience-manager-content-response-1

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 7]
```diff
@@ -4,8 +4,160 @@
...
+
+This event has 2 purposes:
+
+1. Deliver the Audience Manager profile to the original requester
+2. Deliver the Audience Manager profile to any module that may be listening for updates.
+
+This event will be generated in the following scenarios:
+
+- When a request is made to update the Audience manager profile 
+- When a request is made to retrieve the current Audience manager profile
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Instead of adding the same description as in the Audience extension, I would suggest we articulate why is the Identity extension listening for this event, something like:
The Identity extension listens for the Audience extension response content event for any user opt-out updates. If this response contains `optedouthitsent: false` a new opt-out network call is sent to the Identity service.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md (hunk 6 above)
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Use Yes instead of True in all the tables.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 8]
```diff
@@ -4,8 +4,160 @@
...
+
+| Type                                | Source                                | Paired | Direction                                                    |
+| :---------------------------------- | :------------------------------------ | :----- | :----------------------------------------------------------- |
+| com.adobe.eventType.audienceManager | com.adobe.eventSource.responseContent | True   | This is the response to a paired event. This should be published both with and without a paired ID |
+
+#### Data Payload Definition
+
+Definition of the key/value pairs that will be present in this event
+
+| Key + Key Type         | Value Type | Optional | Description                                                  |
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Replace Key + Key Type with `Key`.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 9]
```diff
@@ -4,8 +4,160 @@
...
+| :--------------------- | :--------- | :------- | :----------------------------------------------------------- |
+| <CUSTOMER_KEY>(String) | String     | True     | The Data payload will be full of Customer key value pairs. The Data Payload will be populated using the CV and CN values in the AAM Stuff array |
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: The Key and description should be updated for the optout case.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 10]
```diff
@@ -4,8 +4,160 @@
...
+The data property in the Configuration Response Content event is used by each extension to modify its current settings. Each extension is responsible to read out the part of the data property for which it is concerned.
+
+The Identity extension will read the following keys from the configuration event:
+
+Here are the key-value pairs in this event:
+
+| **Key**                  | **Value Type** | **Optional** | **Description**                                              |
+| :----------------------- | :------------- | :----------- | :----------------------------------------------------------- |
+| `experienceCloud.org`    | String         | Yes          | Experience Cloud Org Identifier                              |
+| `experienceCloud.server` | Sting          | Yes          | Custom endpoint to be used for Visitor ID Service network requests |
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Please include `global.privacy` in the list.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 11]
```diff
@@ -4,8 +4,160 @@
...
+
+
+### Event Hub Shared State
+
+#### Event Description
+
+The shared state event is dispatched by the event hub after a shared state is created or updated. The event data contains the event owner, the name of the module to which the shared state belongs.
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Please include details about why is the Identity extension listening for this event.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 12]
```diff
@@ -211,17 +322,6 @@ The result for the `getUrlVariables` request:
 The Analytics Custom VID is no longer included in the Identity event data payload. If needed, it may be retrieved from the [Analytics extension](https://github.com/Adobe-Marketing-Cloud/aep-sdks-documentation/tree/987092ee0fd2699ae5898755cd72255802628107/using-mobile-extensions/adobe-analytics/analytics-api-reference/README.md#get-the-custom-visitor-identifier).
 {% endhint %}
 
-### Configuration content request
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: I would suggest keeping this order for dispatched events, based on their importance and relation to the Identity extension:
Identity response identity
Configuration Content request
Analytics content request

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 13]
```diff
@@ -126,10 +118,111 @@ Set the Push Identifier request
 }
 ```
 
+### Analytics Identity Response
+
+#### Event Description
+
+This event is dispatched by the Analytics extension upon receiving the Analytics identity request event that gets generated in the Analytics [getTrackingIdentifier](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-analytics/analytics-api-reference#gettrackingidentifier) and [getVisitorIdentifier](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-analytics/analytics-api-reference#getvisitoridentifier-1) calls. This event contains the unique tracking identifier. This value may be null if identity extension is enabled and there is a valid ECID generated.
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Instead of adding the description of the Analytics Identity response event here, I recommend adding a short description about why is the Identity extension listening for this event and adding a link to the https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-analytics/analytics-event-reference#analytics-response-identity

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 14]
```diff
@@ -126,10 +118,111 @@ Set the Push Identifier request
...
+
+```
+{"aid" : "aidValue"}
+```
+
+### Audience Manager Content Response
+
+The Identity extension listens for the Audience extension response content event for any user opt-out updates. If this response contains `optedouthitsent: false` a new opt-out network call is sent to the Identity service.
+
+For the details about this event, please refer the [Audience Manager Event Reference document](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-audience-manager/audience-manager-event-reference#audience-manager-content-response-1).
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Rephrase to: For more details see [Audience Manager Content Response](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-audience-manager/audience-manager-event-reference#audience-manager-content-response-1)

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 15]
```diff
@@ -126,10 +118,111 @@ Set the Push Identifier request
...
+
+### Configuration Response Content
+
+#### Event Description
+
+This event will be generated in the following scenarios:
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Similar here, these details should be covered in the Configuration event reference page and here we should just enumerate what are the keys required by the Identity extension.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md (hunk 12 above)
Comment (emdobrin) [reply using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Let's revert this deletion and keep the following order: 
Identity response identity
Configuration content request
Analytics content request

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 16]
```diff
@@ -126,10 +118,111 @@ Set the Push Identifier request
...
+
+**sample**
+
+```
+{"aid" : "aidValue"}
+```
+
+### Audience Manager Content Response
+
+The Identity extension listens for the Audience extension response content event for any user opt-out updates. If this response contains `optedouthitsent: false` a new opt-out network call is sent to the Identity service.
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: an opt-out network call is sent to the Experience Cloud Identity Service.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md (hunk 12 above)
Comment (samthiru) [reply using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: It is not deleted though 
Ref : https://github.com/samthiru/aep-sdks-documentation/blob/AMSDK-9006/using-mobile-extensions/mobile-core/identity/identity-event-reference.md

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md (hunk 15 above)
Comment (samthiru) [reply using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: For Config, I am just leaving the details of the event as it is ... because https://aep-sdks.gitbook.io/docs/using-mobile-extensions/mobile-core/configuration/configuration-event-reference page is still to be added.

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 17]
```diff
@@ -126,6 +118,75 @@ Set the Push Identifier request
...
+
+For more details see [Audience Manager Content Response Document](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-audience-manager/audience-manager-event-reference#audience-manager-content-response-1).
+
+### Configuration Response Content
+
+#### Event Description
+
+Whenever the configuration extension dispatches the Configuration Response Content event, the Identity extension fetches the URL of the Experience Cloud Identity Service,  Experience Cloud Organization Identifier, and the Privacy status settings.
+
+This event will be generated in the following scenarios:
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: Use present tense for all the sentenses. E.g. This event is generated in the following scenarios:

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 18]
```diff
@@ -126,6 +118,75 @@ Set the Push Identifier request
...
+
+| **Event Type**                    | **Event Source**                      | **Paired** | **Paired Event** |
+| :-------------------------------- | :------------------------------------ | :--------- | :--------------- |
+| com.adobe.eventType.configuration | com.adobe.eventSource.responseContent | No         | N/A              |
+
+The data property in the Configuration Response Content event is used by each extension to modify its current settings. Each extension is responsible to read out the part of the data property for which it is concerned.
+
+#### Data Payload Definition
+
+The Identity extension will read the following keys from the configuration event:
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: The Identity extension reads the following...

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 19]
```diff
@@ -126,6 +118,75 @@ Set the Push Identifier request
...
+
+#### Event Details
+
+| **Event Type**          | **Event Source**                  | **Paired** | **Paired Event** |
+| :---------------------- | :-------------------------------- | :--------- | :--------------- |
+| com.adobe.eventType.hub | com.adobe.eventSource.sharedState | No         | N/A              |
+
+#### Data Payload Definition
+
+Definition of the key/value pairs that will be present in this event
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: pairs that are present in this event:

File: using-mobile-extensions/mobile-core/identity/identity-event-reference.md [hunk 20]
```diff
@@ -126,6 +118,75 @@ Set the Push Identifier request
...
+| `experienceCloud.org`    | String         | Yes          | Experience Cloud Org Identifier                              |
+| `experienceCloud.server` | Sting          | Yes          | Custom endpoint to be used for Visitor ID Service network requests |
+| `global.privacy`         | String         | Yes          | Contains the mobile privacy status settings.                 |
+
+
+### Event Hub Shared State
+
+#### Event Description
+
+The shared state event is dispatched by the event hub after a shared state is created or updated. The event data contains the event owner, the name of the extension to which the shared state belongs. Upon hearing this event, the relevant listener rergistered by the Identity extension triggers processing of Identity module's event queue.
```
Comment (emdobrin) [inline using-mobile-extensions/mobile-core/identity/identity-event-reference.md]: I would rephrase the last sentence to: `The Identity extension triggers processing the queued events, if any, when the Configuration's shared state is updated.`
